### ISSUE TITLE
KerasR for time serie

### ISSUE THREAD

**Issue by dma100180** [statsmaths/kerasR#42]
Hello, I am trying to create an LSTM neural network with KERASR for the prediction of time series with several variables and I can not find any example, do you know any?
Thank you

**Comment by dma100180** [comment]
Hello, following one of the example (https://github.com/rstudio/keras/blob/master/vignettes/examples/stateful_lstm.R), I changed the data and it works for me, but when I only received one input measure, when I try to do it with two I get an error:
Error when checking input: expected lstm_8_input to have 3 dimensions, but got array with shape (1203, 1203, 1, 1)

[Uploading data.zip…]()
DataKeras<-read.csv(file="D:/data.csv")
DataKeras
dak<-DataKeras[,c(3,50)]

trx <- array(data = c(dak[,1],dak[,2]), dim = c(length(dak[,1]),length(dak[,1]), 1, 1))
try <- array(data = dak[,1], dim = c(length(dak[,1]), 1))
dim(trx)
dim(try)

library(keras)
# since we are using stateful rnn tsteps can be set to 1
tsteps <- 1
batch_size <- 3
epochs <- 25
# number of elements ahead that are used to make the prediction
lahead <- 2

model <- keras_model_sequential()
model %>%
  layer_lstm(units = 50, input_shape = c(tsteps,1), batch_size = batch_size,
             return_sequences = TRUE, stateful = TRUE) %>% 
  layer_lstm(units = 50, return_sequences = FALSE, stateful = TRUE) %>% 
  layer_dense(units = 1)
model %>% compile(loss = 'mse', optimizer = 'rmsprop')


for (i in 1:epochs) {
  model %>% fit(trx, try, batch_size = batch_size,
                epochs = 1, verbose = 1, shuffle = FALSE)
  
  model %>% reset_states()
}

predicted_output <- model %>% predict(trx, batch_size = batch_size)


Thank you!!
